tests/test_cxl_image_classification_host.py

```diff
@@ -0,0 +1,156 @@
+"""
+ Copyright (c) 2024, Eeum, Inc.
+
+ This software is licensed under the terms of the Revised BSD License.
+ See LICENSE for details.
+"""
+
+# pylint: disable=unused-import, duplicate-code
+import asyncio
+import pytest
+
+from opencxl.apps.cxl_complex_host import CxlComplexHost, CxlComplexHostConfig
+from opencxl.apps.cxl_image_classification_host import CxlImageClassificationHostConfig
+from opencxl.cxl.component.common import CXL_COMPONENT_TYPE
+from opencxl.cxl.component.root_complex.home_agent import MEMORY_RANGE_TYPE, MemoryRange
+from opencxl.cxl.component.root_complex.root_complex import RootComplexMemoryControllerConfig
+from opencxl.cxl.component.root_complex.root_port_client_manager import RootPortClientConfig
+from opencxl.cxl.component.root_complex.root_port_switch import (
+    COH_POLICY_TYPE,
+    ROOT_PORT_SWITCH_TYPE,
+)
+from opencxl.apps.cxl_host import CxlHostManager
+from opencxl.cxl.component.switch_connection_manager import SwitchConnectionManager
+from opencxl.cxl.component.cxl_component import PortConfig, PORT_TYPE
+from opencxl.cxl.component.physical_port_manager import PhysicalPortManager
+from opencxl.cxl.component.virtual_switch_manager import (
+    VirtualSwitchManager,
+    VirtualSwitchConfig,
+)
+from opencxl.apps.accelerator import MyType1Accelerator, MyType2Accelerator
+from opencxl.drivers.cxl_mem_driver import CxlMemDriver
+from opencxl.util.number_const import MB
+from opencxl.util.logger import logger
+from opencxl.drivers.cxl_bus_driver import CxlBusDriver
+from opencxl.drivers.pci_bus_driver import PciBusDriver
+
+BASE_TEST_PORT = 9300
+
+
+@pytest.mark.asyncio
+async def test_cxl_host_type1_image_classification_host_ete():
+    # pylint: disable=protected-access
+    host_port = BASE_TEST_PORT + pytest.PORT.TEST_5 + 155
+    util_port = BASE_TEST_PORT + pytest.PORT.TEST_5 + 156
+    switch_port = BASE_TEST_PORT + pytest.PORT.TEST_5 + 157
+
+    port_configs = [
+        PortConfig(PORT_TYPE.USP),
+        PortConfig(PORT_TYPE.DSP),
+        PortConfig(PORT_TYPE.DSP),
+        PortConfig(PORT_TYPE.DSP),
+        PortConfig(PORT_TYPE.DSP),
+    ]
+    sw_conn_manager = SwitchConnectionManager(port_configs, port=switch_port)
+    physical_port_manager = PhysicalPortManager(
+        switch_connection_manager=sw_conn_manager, port_configs=port_configs
+    )
+
+    switch_configs = [
+        VirtualSwitchConfig(upstream_port_index=0, vppb_counts=4, initial_bounds=[1, 2, 3, 4])
+    ]
+    virtual_switch_manager = VirtualSwitchManager(
+        switch_configs=switch_configs, physical_port_manager=physical_port_manager
+    )
+
+    dev1 = MyType1Accelerator(port_index=1, port=switch_port, irq_listen_port=9100)
+    dev2 = MyType1Accelerator(port_index=2, port=switch_port, irq_listen_port=9101)
+    dev3 = MyType1Accelerator(port_index=3, port=switch_port, irq_listen_port=9102)
+    dev4 = MyType1Accelerator(port_index=4, port=switch_port, irq_listen_port=9103)
```

Comment:
Please create a loop so that we could easily scale later. e.g. if we want to decrease it to 2 devices.